v2/blockstore/readwrite.go

```diff
@@ -260,7 +260,35 @@ func (b *ReadWrite) Finalize() error {
 }
 
 func (b *ReadWrite) AllKeysChan(ctx context.Context) (<-chan cid.Cid, error) {
-	return b.ronly.AllKeysChan(ctx)
+	if ctx.Err() != nil {
+		return nil, ctx.Err()
+	}
+
+	b.ronly.mu.Lock()
+	defer b.ronly.mu.Unlock()
+
+	if b.ronly.closed {
+		return nil, errClosed
+	}
+
+	out := make(chan cid.Cid)
+
+	go func() {
+		defer close(out)
+		err := b.idx.ForEachCid(func(c cid.Cid, _ uint64) error {
+			select {
```

Comment:
```suggestion
		err := b.idx.ForEachCid(func(c cid.Cid, _ uint64) error {
			if !b.opts.BlockstoreUseWholeCIDs {
				c = cid.NewCidV1(cid.Raw, c.Hash())
			}
			select {
```

Reply:
I think this is all you should need - so we have the rule: if the user requested storage with whole CIDs then they get whole (original) CIDs back from an index listing, if they just wanted multihash like a standard blockstore then they get raw CIDs out of this operation.

Essentially the same behaviour as https://github.com/ipld/go-car/blob/a2a8d2f9f60fba6cb0c0b985d9599bcc03a3e2f4/v2/blockstore/readonly.go#L408-L410